bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnitHandler.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.danfossairunit.internal;
+
+import static org.openhab.binding.danfossairunit.internal.DanfossAirUnitBindingConstants.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DanfossAirUnitHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ralf Duckstein - Initial contribution
+ * @author Robert Bach - heavy refactorings
+ */
+@NonNullByDefault
+public class DanfossAirUnitHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(DanfossAirUnitHandler.class);
+
+    private @Nullable DanfossAirUnitConfiguration config;
+
+    public DanfossAirUnitHandler(Thing thing) {
+        super(thing);
+    }
+
+    private @Nullable ValueCache valueCache;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private @Nullable DanfossHRV hrv;
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAllChannels();
+        } else {
+            if (hrv == null) {
+                return;
+            }
+            try {
+                DanfossHRV danfossHRV = hrv;
+                Channel channel = Channel.getByName(channelUID.getIdWithoutGroup());
+                hrv.connect();
+                if (channel.getWriteAccessor() != null) {
+                    updateState(channelUID, channel.getWriteAccessor().access(danfossHRV, command));
+                }
+            } catch (IllegalArgumentException e) {
```

Comment:
What are you expecting to throw this?

Reply:
I had this kind of exceptions in the past (probably due to concurrency issues as I know now), but I'd like to keep that in place to gracefully handle any communication problems in the "right place". The exception is originally thrown in sendRequest() method (former DanfossHRV class, now DanfossAirUnitCommunicationController).